### PR TITLE
[recnet-api] Add additional fields to user profile

### PR DESCRIPTION
**Pull request by joannechen1223**
## Description

Add additional fields to the User table and enable them in create/update user API:

- url
- googleScholarLink
- semanticScholarLink
- openReviewUserName

## Related Issue

- https://github.com/lil-lab/recnet/issues/76

## Notes

I add these fields in the onboarding page in the frontend to pass the typecheck. If there is anything need to be changed on that part, please let me know.

## TODO

- [ ] Paste the testing link
- [ ] Clear `console.log` or `console.error` for debug usage
- [ ] Update the documentation `recnet-docs` if needed
- [ ] Version bump in `package.json` if needed

### CODE REVIEW

**Comment by swh00tw** [comment]
So these four new fields are not in "UserPreview"? I am not sure if this is the right decision. Even though these fields won't be displayed anywhere for now, I think these fields aren't sensitive and should be included in "UserPreview".

**Comment by joannechen1223** [comment]
> So these four new fields are not in "UserPreview"? I am not sure if this is the right decision. Even though these fields won't be displayed anywhere for now, I think these fields aren't sensitive and should be included in "UserPreview".

In my opinion, I would like to only have the necessary fields in the user preview, and as least as possible. if these fields are not disclosed to other users on the UI, then we should not let them be accessible through the API.

File: apps/recnet-api/src/database/repository/user.repository.ts [hunk 1]
```diff
@@ -113,9 +114,10 @@ export default class UserRepository {
 
   public async updateUser(
     userId: string,
-    updateUserInput: Prisma.UserUpdateInput
+    updateUserInput: UpdateUserInput
   ): Promise<User> {
     const where = { id: userId };
+    console.log(updateUserInput);
```
Comment (swh00tw) [inline apps/recnet-api/src/database/repository/user.repository.ts:120]:
```suggestion
```

**Review by swh00tw — APPROVED** [review]

**Comment by swh00tw** [comment]
> > So these four new fields are not in "UserPreview"? I am not sure if this is the right decision. Even though these fields won't be displayed anywhere for now, I think these fields aren't sensitive and should be included in "UserPreview".
> 
> In my opinion, I would like to only have the necessary fields in the user preview, and as least as possible. if these fields are not disclosed to other users on the UI, then we should not let them be accessible through the API.

Make sense. Originally, I was thinking maybe at some point we would display these new fields in Profile or somewhere. So that put them in "UserPreview" would be better. I guess we could make changes if we decide to do it in the future. Looks good right now.

**Pull request merged**